### PR TITLE
Rename `LOGOUT_WPCOM` to `REMOVE_WPCOM_SITES`

### PR DESCRIPTION
**Pull request by maxme**
`LOGOUT_WPCOM` was a bad name because: 1. we have a `SIGNOUT` action in AccountAction 2. It's not obvious what happen in the SiteStore when we dispatch it.

`REMOVE_WPCOM_SITES` seems like a better name.

### CODE REVIEW

File: example/src/test/java/org/wordpress/android/stores/SiteStoreUnitTest.java [hunk 1]
```diff
@@ -281,7 +281,7 @@ public void testGetRestApiSites() {
         SiteSqlUtils.insertOrUpdateSite(jetpackSiteOverDotOrg);
         SiteSqlUtils.insertOrUpdateSite(jetpackSiteOverRestOnly);
 
-        List<SiteModel> restApiSites = SiteSqlUtils.getAllRestApiSites();
+        List<SiteModel> restApiSites = SiteSqlUtils.getAllWPComSites();
```
Comment (aforcier) [inline example/src/test/java/org/wordpress/android/stores/SiteStoreUnitTest.java]: Might as well rename the entire method to `testGetWPComSites`, and `restApiSites` to `wpcomSites`,  to completely eliminate `restApiSite`-type names from the codebase.

Comment (maxme) [inline example/src/test/java/org/wordpress/android/stores/SiteStoreUnitTest.java]: Names updated in  	c44e769

**Comment by aforcier** [comment]
:shipit:

**Comment by maxme** [comment]
Thanks for d2ebe38 !

**Pull request merged**